### PR TITLE
[test] SR-1238: Add test for UnsafePointer and UnsafeMutablePointer conformance to Comparable

### PR DESCRIPTION
**Pull request by JaviSoto**
#### What's in this pull request?

Added tests for `Comparable` conformance of `UnsafePointer` and `UnsafeMutablePointer`
#### Resolved bug number: ([SR-1238](https://bugs.swift.org/browse/SR-1238))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

 **Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| OS X platform | @swift-ci Please test OS X platform |
| Linux platform | @swift-ci Please test Linux platform |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by gribozavr** [comment]
@swift-ci Please test

File: test/1_stdlib/UnsafePointer.swift.gyb [hunk 1]
```diff
@@ -389,5 +389,26 @@ UnsafeMutablePointerTestSuite.test("customPlaygroundQuickLook") {
...
+
+${SelfName}TestSuite.test("Comparable") {
+  let instances = [
+    0x00000001,
+    0xFF00FF00,
+    0x00001111,
+    0x00000002,
+    0xFFFFFFFF,
+    0x00001111
+  ].map { ($0, ${SelfType}(bitPattern: $0)!) }
```
Comment (gribozavr) [inline test/1_stdlib/UnsafePointer.swift.gyb]: It would be good to throw in some 64-bit values if we are on a 64-bit platform (see #ifs in tests above).

Comment (JaviSoto) [inline test/1_stdlib/UnsafePointer.swift.gyb]: 👍 will do

**Comment by JaviSoto** [comment]
AFAICT CI failed with an unrelated failure. I pushed a new commit with 64 bit values as requested however.

**Comment by gribozavr** [comment]
@JaviSoto Thank you!

@swift-ci Please test and merge

**Comment by jrose-apple** [comment]
Should AutoreleasingUnsafeMutablePointer be included? It's also Strideable.

Should OpaquePointer be included? It's *not* Strideable, but it probably still makes sense to be Comparable.

**Comment by gribozavr** [comment]
@jrose-apple Thank you, we are forgetting about other pointer types again!  I think it makes sense for them to be comparable.

**Comment by JaviSoto** [comment]
I'll open a separate PR for those!

**Pull request merged**